code/game/objects/items/implants/implant_blocker.dm

```diff
@@ -0,0 +1,27 @@
+/obj/item/implant/blocker
+	name = "blocker implant"
+	desc = "Injects things."
+	icon_state = "reagents"
+	var/block_mode = 0
+	var/block_arg = list("Assistant")
+	activated = FALSE
+	var/datum/component/C
+
+/obj/item/implant/blocker/implant(mob/living/target)
+	. = ..()
+	C = target.AddComponent(/datum/component/blocker, block_mode, block_arg) // i know this probably isn't the right way to do this, i'll fix it tomorrow
```

Comment:
Works fine enough. I would probably add some sanity in case the AddComponent fails and returns null.